src/python/pants/backend/docker/target_types.py

```diff
@@ -333,6 +334,55 @@ def option_values(self, **kwargs) -> Iterator[str]:
         yield from cast("tuple[str]", self.value)
 
 
+class DockerBuildUnaryOptionFieldMixin(Field):
+    """Inherit this mixin class to provide unary options (i.e. option in the form of `--flag=value`)
+    to `docker build`."""
+
+    docker_build_option: ClassVar[str]
+
+    @final
+    def options(self) -> Iterator[str]:
+        yield f"{self.docker_build_option}={self.value}"
+
+
+class DockerImageBuildPullOptionField(DockerBuildUnaryOptionFieldMixin, BoolField):
+    alias = "pull"
+    default = True
+    help = softwrap(
+        """
+        If true, then docker will always attempt to pull a newer version of the image.
+
+        Useful to disable it when building images from other intermediate goals.
+        """
+    )
+    docker_build_option = "--pull"
+
+
+class DockerBuildSkippingOptionFieldMixin(BoolField, ABC):
```

Comment:
The doc string gives a better description than the class name imo with the `optional` keyword, so how about renaming this to something like `DockerOptionalBuildOptionFieldMixin` or even `DockerBuildOptionFlagFieldMixin`.
 
I can see that optional and option can trip you up a bit here, but I feel optional better convey the use over skipping. (I associated this with an option for skipping the build, not the option for instance)

Reply:
I'm gonna go with `DockerBuildOptionFlagFieldMixin`, to keep `DockerBuild...` naming convention